### PR TITLE
Update test dependencies and fix github actions cache.

### PR DESCRIPTION
**Pull request by apollo13**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -27,15 +27,15 @@ jobs:
       - uses: actions/cache@v3
         with:
           path: ~/.cache/pip
-          key: ${{ runner.os }}-pip-${{ matrix.python }}-${{ hashFiles('setup.cfg') }}
+          key: ${{ runner.os }}-pip-${{ matrix.python }}-${{ hashFiles('pyproject.toml') }}
```
Comment (apollo13) [inline .github/workflows/test.yml:30]: Forgot to change that before, now `setup.cfg` is static and rather useless.

File: pyproject.toml [hunk 2]
```diff
@@ -54,7 +54,7 @@ default_section = "THIRDPARTY"
 known_first_party = "goodconf"
 
 [tool.pytest.ini_options]
-addopts = "--cov"
+addopts = "--cov=goodconf --cov-branch"
```
Comment (apollo13) [inline pyproject.toml:57]: Cover only `goodconf` instead of tests as well.

Comment (ipmb) [reply pyproject.toml:57]: Having coverage on tests will show you if a test you wrote isn't being executed for some reason. Wouldn't that be helpful?

Comment (apollo13) [reply pyproject.toml:57]: Mhm good point, will push (although depending on whether features are activated via python version we might have to combine coverage output at some point :D)

**Pull request merged**